thefuck/rules/git_push_different_branch_names.py

```diff
@@ -0,0 +1,19 @@
+import re
+from thefuck.utils import replace_argument
+from thefuck.specific.git import git_support
+
+
+@git_support
+def match(command):
+    return "git push" in command.script and "The upstream branch of your current branch does not match" in command.output
+
+
+@git_support
+def _parse_git_output(output):
+    regex = re.compile(r'^ +(git push [^\s]+ [^\s]+)', re.MULTILINE)
+    return regex.findall(output)[0]
```

Comment:
These 2 lines can be replaced by one:

```python
return re.findall(r'^ +(git push [^\s]+ [^\s]+)', command.output, re.MULTILINE)
```

And maybe move it inside `get_new_command`, while **fixing** the latter too.